.github/workflows/lint_build_test.yaml

```
@@ -0,0 +1,164 @@
+name: Lint and Build
+
+on:
+  pull_request:
+
+env:
+  NODE_VERSION: 10
+  GO_VERSION: 1.15
+  JAVA_VERSION: 9
+
+jobs:
+  javascript:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+
+      - name: JavaScript - kyber
+        run: |
+          cd external/js/kyber
+          npm ci
+          npm run linter
+          npm run build
+
+      - name: JavaScript - cothority
+        run: |
+          cd external/js/cothority
+          npm ci
+          npm run linter
+          npm link @dedis/kyber
+          npm run build
+
+  golang:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        golang-version: [1.14, 1.15] # Cannot use env. here :(
+
+    steps:
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 1
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.golang-version }}
+
+      - name: Installing golint
+        run: GO111MODULE=off go get -u golang.org/x/lint/golint
+
+      - name: Testing proto-files
+        run: |
+          make test_proto
+          go version
+
+      - name: Running make-tests
+        run: |
+          make -C conode verify
+          make test_{fmt,lint}
+
+  # As pre-commit doesn't like to be run in parallel, this step runs it only
+  # for the latest go-version.
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 1
+      - uses: actions/setup-python@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Installing golint
+        run: GO111MODULE=off go get -u golang.org/x/lint/golint
+
+      - name: Running pre-commit
+        uses: pre-commit/action@v2.0.0
+
+  # The barrier job is used to avoid writing a lot of needs
+  barrier:
```

```suggestion
  lint-and-build:
```

But it's not really correct, as `golang` doesn't build. Consider splitting it into a `lint` and another `build` barrier-job. Also rename the "javascript" & "golang" in "lint-javascript" & "lint-golang".

---

Hmm - currently the `javascript` target lints and builds the code. So _should_ the `golang` target. I'll push a version with that proposition.